test/Router/IntegrationTest.php

```diff
@@ -73,9 +74,11 @@ public function createApplicationFromRouter(RouterInterface $router): Applicatio
     /**
      * Get the router adapters to test
      *
-     * @psalm-return iterable<class-string>
+     * @psalm-return Generator<int, array{
+     *     0: (AuraRouter::class|FastRouteRouter::class|LaminasRouter::class)
+     * }, mixed, void>
```

Comment:
I never saw this `(ClassOne::class|ClassTwo::class)` syntax but I think this is more extensible:

```suggestion
     * @psalm-return Generator<int, array{class-string<RouterInterface>}>
```

Also, AFAIK `Generator<int, string>` is a short-cut for `Generator<int, string, mixed, mixed>` and we don't really care about types for the generator's input nor it's return.

Reply:
Thanks for the suggestion, I will push that change. 

Initially I changed that when I ran `vendor/bin/psalm --alter --issues=InvalidReturnType`.